database/seeders/RequestedSongSeeder.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Database\Seeders;
+
+use App\Models\Company;
+use App\Models\RequestedSong;
+use App\Models\Song;
+use App\Models\User;
+use Illuminate\Database\Seeder;
+
+class RequestedSongSeeder extends Seeder
+{
+  public function run(): void
+  {
+    for ($i = 0; $i < 15; $i++) {
+      RequestedSong::create([
+        'song_id' => Song::all()->take(5)->random()->id,
+        'user_id' => User::where('role', 2)->get()->random()->id,
+        'company_id' => Company::all()->random()->id
```

Comment:
- Takes the queries out.
- You can take in random order : https://laravel.com/docs/10.x/queries#random-ordering
- Use limit() to reduce the volume of taken songs
- Take only songs that are not blacklisted

Reply:
Took out all the queries. Make a `DB query` to get all the songs that have not been blacklisted. I continue to `random()` as it is one of the methods of Laravel Collections